src/clj/fluree/db/db/json_ld.cljc

```diff
@@ -373,7 +397,62 @@
     (jld-format/forward-properties db iri spec context compact-fn cache fuel-tracker error-ch))
 
   (-reverse-property [db iri reverse-spec compact-fn cache fuel-tracker error-ch]
-    (jld-format/reverse-property db iri reverse-spec compact-fn cache fuel-tracker error-ch)))
+    (jld-format/reverse-property db iri reverse-spec compact-fn cache fuel-tracker error-ch))
+
+  (-iri-visible? [db iri]
+    (let [sid (iri/encode-iri db iri)]
+      (validate/allow-iri? db sid)))
+
+  indexer/Indexed
+  (collect [db changes-ch]
+    (if (idx-default/novelty-min? db reindex-min-bytes)
+      (idx-default/refresh db changes-ch)
+      (go)))
+
+
+  TimeTravel
```

Comment:
I'm liking these new protocols a lot. This looks like a nice step towards decomplecting : )